src/WebJobs.Script.WebHost/Swagger/SwaggerDocumentManager.cs

```diff
@@ -0,0 +1,168 @@
+{
+    public class SwaggerDocumentManager : ISwaggerDocumentManager
+    {
+        private readonly string _swaggerFilePath;
+        private readonly string _swaggerDirPath;
+        private readonly ScriptHostConfiguration _config;
+
+        public SwaggerDocumentManager(ScriptHostConfiguration hostConfig)
+        {
+            _config = hostConfig;
+            _swaggerDirPath = Path.Combine(_config.RootScriptPath, ScriptConstants.AzureFunctionsSystemDirectoryName, ScriptConstants.SwaggerDirectoryName);
+            _swaggerFilePath = Path.Combine(_swaggerDirPath, ScriptConstants.SwaggerFileName);
+        }
+
+        public async Task<JObject> GetSwaggerDocumentAsync() => await ReadSwaggerAsync();
+
+        public JObject GenerateSwaggerDocument(IDictionary<IHttpRoute, FunctionDescriptor> httpFunctions)
+        {
+            var swaggerDocument = new SwaggerDocument();
+            string hostname = ScriptSettingsManager.Instance.GetSetting(EnvironmentSettingNames.AzureWebsiteHostName);
+            swaggerDocument.SwaggerInfo.FunctionAppName = hostname;
+            swaggerDocument.Host = hostname;
+            swaggerDocument.ApiEndpoints = GetEndpointsData(httpFunctions);
+            return JObject.FromObject(swaggerDocument);
+        }
+
+        private static Dictionary<string, Dictionary<string, HttpOperationInfo>> GetEndpointsData(IDictionary<IHttpRoute, FunctionDescriptor> httpFunctions)
+        {
+            var apiEndpoints = new Dictionary<string, Dictionary<string, HttpOperationInfo>>();
+            foreach (var httpRoute in httpFunctions.Keys)
+            {
+                if (httpFunctions[httpRoute].Metadata.IsDisabled == true)
+                {
+                    continue;
+                }
+
+                string endpoint = $"/{httpRoute.RouteTemplate}";
+                Dictionary<string, HttpOperationInfo> endpointsOperationData;
+                if (!apiEndpoints.TryGetValue(endpoint, out endpointsOperationData))
+                {
+                    endpointsOperationData = new Dictionary<string, HttpOperationInfo>();
+                    apiEndpoints.Add(endpoint, endpointsOperationData);
+                }
+
+                string[] httpOperations = GetHttpOperations(httpRoute);
+                Collection<HttpOperationParameterInfo> inputParameters = GetInputParameters(httpRoute);
+                foreach (var httpOperation in httpOperations)
+                {
+                    // Trace is not recogized as an HttpMethod by swagger specification
+                    if (httpOperation.ToLower(CultureInfo.CurrentCulture) == HttpMethod.Trace.ToString().ToLowerInvariant())
+                    {
+                        continue;
+                    }
+
+                    HttpOperationInfo httpOperationInfo = new HttpOperationInfo()
+                    {
+                        InputParameters = inputParameters,
+                        OperationId = $"{endpoint}/{httpOperation}"
```

Comment:
OperationId could be anything. No specific format to it. Strings are constructed both here and above because of the way the sources have it formatted. Depending on the source this could change.